### PR TITLE
Secrets, MQTT interface, interface & protocol cmds

### PR DESCRIPTION
**Pull request by ryanmelt**

### CODE REVIEW

File: openc3/lib/openc3/utilities/secrets.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# This file may also be used under the terms of a commercial license
+# if purchased from OpenC3, Inc.
+
+ENV['OPENC3_SECRET_BACKEND'] ||= 'redis'
+
+module OpenC3
+  class Secrets
```
Comment (jmthomas) [inline openc3/lib/openc3/utilities/secrets.rb:22]: I think this API will work well with the methods defined by https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/SecretsManager/Client.html

File: openc3/lib/openc3/models/interface_model.rb [hunk 2]
```diff
@@ -242,6 +256,14 @@ def handle_config(parser, keyword, parameters)
         parser.verify_num_parameters(0, 0, "#{keyword}")
         @log_raw = true
 
+      when 'SECRET'
+        parser.verify_num_parameters(3, 4, "#{keyword} <Secret Type: ENV or FILE> <Secret Name> <Environment Variable Name or File Path> <Option Name (Optional)>")
+        @secrets << parameters[0..2]
+        if parameters[3]
+          # Option Name, Secret Name
+          @secret_options << [parameters[3], parameters[1]]
+        end
```
Comment (jmthomas) [inline openc3/lib/openc3/models/interface_model.rb:265]: You can make this a secret and a secret option at the same time? What does that look like?

**Review by jmthomas — APPROVED** [review]

**Comment by jmthomas** [comment]
closes #269

File: openc3/lib/openc3/models/interface_model.rb (hunk 2 above)
Comment (ryanmelt) [reply openc3/lib/openc3/models/interface_model.rb:265]: See the example openc3-cosmos-mqtt-test plugin I added.

**Pull request merged**